### PR TITLE
#6583: Fix acosh_bw

### LINKED ISSUE #6583: [Bug Report] invalid acosh backward result
**Describe the bug**

The `acosh_bw` function returns an invalid gradient value.

**To Reproduce**
1. Copy and past below code to /tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_acosh.py 
```Python
# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.

# SPDX-License-Identifier: Apache-2.0

import torch
import pytest
import tt_lib
from tests.tt_eager.python_api_testing.unit_testing.backward_ops.utility_funcs import data_gen_pt_tt, compare_results

def data_gen_pt_tt(input_shapes, device, required_grad=False, val=1):
    pt_tensor = (torch.ones(input_shapes, requires_grad=required_grad) * val).bfloat16()
    tt_tensor = (
        tt_lib.tensor.Tensor(pt_tensor, tt_lib.tensor.DataType.BFLOAT16).to(tt_lib.tensor.Layout.TILE).to(device)
    )
    return pt_tensor, tt_tensor

@pytest.mark.parametrize(
    "input_shapes",
    (
        (torch.Size([1, 1, 32, 32])),
    ),
)
def test_bw_acosh(input_shapes, device):
    in_data, input_tensor = data_gen_pt_tt(input_shapes, device, True, val=0.5)
    grad_data, grad_tensor = data_gen_pt_tt(input_shapes, device, False, val=1)

    print("input_tensor", input_tensor)
    print("grad_tensor", grad_tensor)
    
    pyt_y = torch.acosh(in_data)

    tt_output_tensor_on_device = tt_lib.tensor.acosh_bw(grad_tensor, input_tensor)

    in_data.retain_grad()

    pyt_y.backward(gradient=grad_data)

    golden_tensor = [in_data.grad]

    comp_pass = compare_results(tt_output_tensor_on_device, golden_tensor)
    
    print("tt_output_tensor_on_device", tt_output_tensor_on_device)
    print("golden_tensor", golden_tensor)
    assert comp_pass

```
2. Run cmd `pytest ./tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_acosh.py
```Python
input_tensor ttnn.Tensor([[[[ 0.50000,  0.50000,  ...,  0.50000,  0.50000],
               [ 0.50000,  0.50000,  ...,  0.50000,  0.50000],
               ...,
               [ 0.50000,  0.50000,  ...,  0.50000,  0.50000],
               [ 0.50000,  0.50000,  ...,  0.50000,  0.50000]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
grad_tensor ttnn.Tensor([[[[ 1.00000,  1.00000,  ...,  1.00000,  1.00000],
               [ 1.00000,  1.00000,  ...,  1.00000,  1.00000],
               ...,
               [ 1.00000,  1.00000,  ...,  1.00000,  1.00000],
               [ 1.00000,  1.00000,  ...,  1.00000,  1.00000]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
tt_output_tensor_on_device [ttnn.Tensor([[[[inf     , inf     ,  ..., inf     , inf     ],
               [inf     , inf     ,  ..., inf     , inf     ],
               ...,
               [inf     , inf     ,  ..., inf     , inf     ],
               [inf     , inf     ,  ..., inf     , inf     ]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)]
golden_tensor [tensor([[[[nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan],
          ...,
          [nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan]]]], dtype=torch.bfloat16)]
```

**Expected behavior**

I want `acosh_bw` to return the correct gradient

**Screenshots**

**Please complete the following environment information:**
- OS: [e.g. Ubuntu 20.04]
- Version of software (eg. commit 2c776dbd4f4fe9e26fa1e65216db3690a8a800c8)

**Additional context**

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/6583

### Problem description
Acosh backward fails for edge cases when input, gradient is in range [0,1]

### What's changed

- Update `acosh_bw` Logic to match PyTorch API
- As mentioned [here](https://github.com/tenstorrent/tt-metal/pull/11243#discussion_r1808816083), used nan_to_num to compare Nan to numbers.
  - Test passes now in both BH , WH

Updated acosh backward logic vs current logic is as below :

Screenshot 2025-03-27 at 6 32 02 PM

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/14134224476) CI passes
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/runs/14109838293) CI passes as in main
- [x] https://github.com/tenstorrent/tt-metal/actions/runs/14164235499 CI passes for acosh_bw sweep test
- [x] https://github.com/tenstorrent/tt-metal/actions/runs/14164230219 CI passes for acosh_bw sharded sweep test

### CODE REVIEW

**Review by Aswinmcw — APPROVED** [review]
LGTM

**Review by umadevimcw — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp [hunk 1]
```diff
@@ -641,24 +641,41 @@ std::vector<Tensor> ExecuteUnaryBackwardCos::invoke(
...
+                std::nullopt,
+                output_mem_config),
+            ttnn::multiply(
+                ttnn::add(
+                    check_condition, ttnn::eqz(check_condition, output_mem_config), std::nullopt, output_mem_config),
+                t_inf,
+                std::nullopt,
+                output_mem_config),
+            grad_a,
+            output_mem_config),
```
Comment (sjameelTT) [inline ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp:677]: Since this fixes a bug, and the previous implementation was already a big composite, this is fine. I'm not sure why we don't just support the OP natively rather than making large composites like this.

**Review by sjameelTT — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp (hunk 1 above)
Comment (VirdhatchaniKN) [reply ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp:677]: We have a plan to work on moving the composite ops as device later. Will do that as a part of that task.

**Review by mcw-anasuya — APPROVED** [review]

**Review by omilyutin-tt — APPROVED** [review]

**Review by dchenTT — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/14164196134 passed

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/14166584644 - passed
